### PR TITLE
h00gr shell variables cheat sheet

### PR DESCRIPTION
**Pull request by Ray-Hoo**
**What does your Instant Answer do?**
shell variable cheat sheet

**What is the data source for your Instant Answer? (Provide a link if possible)**
http://bash.cyberciti.biz/guide/Variables

**Why did you choose this data source?**
most complete

**Are there any other alternative (better) data sources?**
Not to my knowledge

**Which communities will this Instant Answer be especially useful for? (gamers, book lovers, etc)**
Linux system administrators

**Is this Instant Answer connected to a DuckDuckHack [Instant Answer idea](https://duck.co/ideas)?**
Yes

**Are you having any problems? Do you need our help with anything?**
No

**Where did you hear about DuckDuckHack? (For first time contributors)**
preemeijer

**What does the Instant Answer look like? (Provide a screenshot for new or updated Instant Answers)**
shell variables cheat sheet

https://duck.co/ia/view/shell_variables_cheat_sheet

### CODE REVIEW

**Comment by jagtalon** [comment]
Hey, @h00gr! Could you link to the idea page? :)

File: share/goodie/cheat_sheets/json/shellvar.json [hunk 1]
```diff
@@ -0,0 +1,102 @@
+{
+    "id": "shell_variables_cheat_sheet",
+    "name": "Shell Variables",
+    "description": "Shell Variables",
+    "metadata": {
+        "sourceName": "Variables",
+        "sourceUrl": "http://bash.cyberciti.biz/guide/Variables"
+    },
+       "aliases": [
+       "basic shell variables"
```
Comment (jagtalon) [inline share/goodie/cheat_sheets/json/shellvar.json]: Maybe "shell variables" would be a good alias as well. This is a bit long

Comment (jagtalon) [inline share/goodie/cheat_sheets/json/shellvar.json]: Also "shell var" and "shell vars"

**Comment by preemeijer** [comment]
@jagtalon he went home but I can provide you the link: https://duck.co/ideas/idea/6202/variables-in-shell-language-cheat-sheet

File: share/goodie/cheat_sheets/json/shellvar.json (hunk 1 above)
Comment (preemeijer) [inline share/goodie/cheat_sheets/json/shellvar.json]: @jagtalon I will ask him but I cant get in to contact with him right now

**Comment by jagtalon** [comment]
@preemeijer thank you! Edited the IA Page: https://duck.co/ia/view/shell_variables_cheat_sheet

File: share/goodie/cheat_sheets/json/shellvar.json (hunk 1 above)
Comment (jagtalon) [inline share/goodie/cheat_sheets/json/shellvar.json]: @preemeijer no worries there's no need to hurry :)

Comment (h00gr) [inline share/goodie/cheat_sheets/json/shellvar.json]: added "shell var", "shell vars" and "shell variables"

**Comment by h00gr** [comment]
I cannot edit that IA page, as I do not have an account there... Where can 
I apply for one?

**Comment by MariagraziaAlastra** [comment]
Hey @h00gr, just go to https://duck.co and sign up there :) then let us know what's your username there and we'll give you permissions for your IA Page. Thanks!

**Comment by h00gr** [comment]
I've just created an account there as hgrdk . Thanks for the support!

**Comment by preemeijer** [comment]
I just want to say: welcome to the DDG community family :+1:

**Comment by MariagraziaAlastra** [comment]
Thanks @h00gr, now you should be able to edit the page: https://duck.co/ia/view/shell_variables_cheat_sheet
Let me know if you have any issues :)

File: share/goodie/cheat_sheets/json/shellvar.json [hunk 2]
```diff
@@ -0,0 +1,102 @@
+{
+    "id": "shell_variables_cheat_sheet",
+    "name": "Shell Variables",
+    "description": "Shell Variables",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/shellvar.json]: Let's remove this, it’s a but redundant :)

File: share/goodie/cheat_sheets/json/shellvar.json [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+    "metadata": {
+        "sourceName": "Variables",
+        "sourceUrl": "http://bash.cyberciti.biz/guide/Variables"
+    },
+       "aliases": [
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/shellvar.json]: Please make sure your indentation is consistent. We use soft-tabs, 4 spaces.

File: share/goodie/cheat_sheets/json/shellvar.json [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+       "basic shell variables", "shell variables", "shell var", "shell vars"
+   ],
+    "template_type": "reference",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/shellvar.json]: Let's use the `terminal` template here :+1:

**Comment by moollaza** [comment]
@h00gr thanks a lot for contributing, we really appreciate it! I've left you some feedback that needs to be implemented. Once things are updated please let me so so I can review this again. If you have any questions, feel free to ask them here!

File: share/goodie/cheat_sheets/json/shellvar.json (hunk 2 above)
Comment (h00gr) [inline share/goodie/cheat_sheets/json/shellvar.json]: Description has been removed.

File: share/goodie/cheat_sheets/json/shellvar.json (hunk 3 above)
Comment (h00gr) [inline share/goodie/cheat_sheets/json/shellvar.json]: corrected this

File: share/goodie/cheat_sheets/json/shellvar.json (hunk 4 above)
Comment (h00gr) [inline share/goodie/cheat_sheets/json/shellvar.json]: changed to the terminal template

**Comment by h00gr** [comment]
Update has been pushed. So please review it again.

File: share/goodie/cheat_sheets/json/shellvar.json [hunk 5]
```diff
@@ -0,0 +1,101 @@
+{
+    "id": "shell_variables_cheat_sheet",
+    "name": "Shell Variables",
+    "metadata": {
+    "sourceName": "Variables",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/shellvar.json]: These ketys/values should be properly indented just like the `sections` are below

**Comment by moollaza** [comment]
@h00gr this LGTM :+1:

Thanks a lot for contributing (we really appreciate it :smiley:). We invite you to stick around and comment/help on other PR's & Issues or even submit an Instant Answer!

**Comment by moollaza** [comment]
@h00gr congrats! It's live: https://duckduckgo.com/?q=shell%20variables%20cheat%20sheet&ia=cheatsheet

Thanks again for taking the time to contribute! We really appreciate it. Feel free to stick around and comment/help on other PR's, tackle open issues or even submit more Instant Answers!

**Pull request merged**